openwillis/measures/audio/util/phonation_util.py

```diff
@@ -0,0 +1,105 @@
+# author:    Georgios Efstathiadis
+# website:   http://www.bklynhlth.com
+
+# author:    Vijay Yadav
+# website:   http://www.bklynhlth.com
+
+# import the required packages
+
+import pandas as pd
+import numpy as np
+import logging
+
+import string
+
+logging.basicConfig(level=logging.INFO)
+logger=logging.getLogger()
+
+PHONATIONS = {
+    'a': ['ah', 'uh', 'huh'],
+    'e': ['eh'],
+    'i': [],
+    'o': ['oh'],
+    'u': [],
+    'm': ['mm', 'hmm', 'hm', 'um']
+}
+
+
+def extract_phonation(speaker_df):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    Extracts phonation samples for each speaker.
+
+    Parameters:
+    ----------
+    speaker_df : pandas DataFrame
+        The speaker data.
+
+    Returns:
+    -------
+    phonation_df : pandas DataFrame
+        The phonation data.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    speaker_df_clean = speaker_df.copy()
+    speaker_df_clean['content'] = speaker_df_clean['content'].apply(lambda x: x.translate(str.maketrans('', '', string.punctuation)).lower())
+
+    phonation_df = pd.DataFrame()
+    for phonation in PHONATIONS:
+        phonation_list = PHONATIONS[phonation]
```

Comment:
eception handling

Reply:
I dont think we need handling here, because this can't really fail on the item level in the loop. And if it does we want it to raise an error and it is handled above where we then have no phonations identified